drivers/counter/Kconfig.alif

```
@@ -0,0 +1,11 @@
+# Copyright (c) 2025 Alif Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config COUNTER_ALIF_UTIMER
+	bool "ALIF Universal Timer Counter driver"
+	default y
```

This will always make it enabled and will cause compilation error if DT_HAS_ALIF_UTIMER_ENABLED is false/unselected. You can achieve what you explained by doing this:
```suggestion
	default y if DT_HAS_ALIF_UTIMER_ENABLED
```